onnx/defs/function.cc

```diff
@@ -93,8 +93,11 @@ Common::Status FunctionBuilderRegistry::GetFunctions(
   }
 
 #ifndef __ONNX_DISABLE_STATIC_REGISTRATION
-  static bool functionBuilder_registerer =
-      (RegisterOnnxFunctionBuilder(), false);
+  static bool registred = false;
+  if (!registred) {
```

Comment:
If the static variable is not used, it might get eliminated. I could change the code back to the original but add a runtime check of `functionBuilder_registerer`

Otherwise `std::mutex` and `std::lock` can be used to make it thread safe

Reply:
No compiler won't eliminate it since it has side effects. static variable itself is thread safe, but now you have changed the "writer" to be out of the static variable's initialization so it's not thread safe anymore. Using lock here should work, but IMO here wrapping/changing RegisterOnnxFunctionBuilder to a class ctor would be simpler. And for the compiler's unused variable warning, you can use the ONNX_UNUSED macro to mark it to silent the compiler.

Reply:
To keep this PR simple I changed it back and used `ONNX_UNUSED`